packages/next/build/index.ts

```diff
@@ -961,3 +970,33 @@ export default async function build(dir: string, conf = null): Promise<void> {
 
   await telemetry.flush()
 }
+
+function generateClientSsgManifest(
+  prerenderManifest: PrerenderManifest,
+  {
+    buildId,
+    distDir,
+    isModern,
+  }: { buildId: string; distDir: string; isModern: boolean }
+) {
+  const ssgPages: Set<string> = new Set<string>([
+    ...Object.keys(prerenderManifest.routes),
+    ...Object.keys(prerenderManifest.dynamicRoutes),
+  ])
```

Comment:
```suggestion
  const dynamicSsgPages = Object.keys(prerenderManifest.dynamicRoutes)
  const ssgPages: Set<string> = new Set<string>([
    ...Object.keys(prerenderManifest.routes)
      .filter(route =>
        // filter out any routes from dynamic SSG pages since we
        // only need the href values to check against
        !dynamicSsgPages.some(dynamicPage =>
          dynamicPage === prerenderManifest.routes[route].srcRoute
        )
      ),
    ...dynamicSsgPages,
  ])
```
We can reduce the size of the `ssgManifest` by removing any routes from a dynamic page's `getStaticPaths` since we only need the `href` values right?

Reply:
Yes, I had this noted as a follow up optimization but I’ll make it now.

Reply:
done